fix(ci): vr-baseline-update workflow fails with 'not in a git directory' in container

## Context

Discovered during #1545 end-to-end smoke test on PR #1537. The bot identity and secret are correctly configured, baseline regeneration succeeds, but the **Commit and push baselines** step fails immediately:

```text
fatal: not in a git directory
Error: Process completed with exit code 128.
```

**Root cause:** Two compounding issues inside the `mcr.microsoft.com/playwright:v1.59.1-noble` container:

1. `actions/checkout@v6` writes its `safe.directory` config under a **temporary HOME** (`/__w/_temp/...`). When the step ends, HOME reverts to `/github/home` and the safe-directory entry is lost.
2. `git config user.name` uses `--local` scope by default, which requires both a valid `.git/` directory and an owner-accepted `safe.directory`. Without the safe-directory entry, git refuses to operate.

The fix is a one-line defensive safe-directory re-registration + explicit `working-directory` pin on the failing step.

**PRD:** `docs/prd/visual-regression-testing.md`
**Umbrella:** #1488
**Sibling:** #1545 (bot identity setup — bot is fully working; only this workflow step is broken)

## Acceptance criteria

- [ ] Add `working-directory: ${{ github.workspace }}` to the **Commit and push baselines** step in `.github/workflows/vr-baseline-update.yml`.
- [ ] Add `git config --global --add safe.directory "${GITHUB_WORKSPACE}"` as the first command in that step's `run:` block.
- [ ] End-to-end smoke test: comment `@kcvv-bot update-vr-baselines` on a PR with a failing `visual-regression` check → bot commits regenerated baselines → CI re-runs and goes green.
- [ ] `pnpm --filter @kcvv/web check-all` passes (no app code changes; workflow-only fix).

## Scope

**Files:** `.github/workflows/vr-baseline-update.yml` only.

--- a/.github/workflows/vr-baseline-update.yml
+++ b/.github/workflows/vr-baseline-update.yml
@@ -136,10 +136,14 @@ jobs:
         run: pnpm --filter @kcvv/web run vr:ci:update
 
       - name: Commit and push baselines
+        working-directory: ${{ github.workspace }}
         env:
           BOT_USER: kcvv-vr-bot
           BOT_EMAIL: kcvv-vr-bot@users.noreply.github.com
         run: |
+          # actions/checkout writes safe.directory under a temp HOME that is
+          # discarded after the step — re-register so git accepts the workspace.
+          git config --global --add safe.directory "${GITHUB_WORKSPACE}"
           git config user.name "$BOT_USER"
           git config user.email "$BOT_EMAIL"
           git add apps/web/test/vr/__snapshots__

